src/main/java/net/neoforged/neoforge/common/extensions/IItemExtension.java

```diff
@@ -76,10 +75,11 @@ default ItemAttributeModifiers getDefaultAttributeModifiers(ItemStack stack) {
     /**
      * Called to allow items to modify the attributes on them.
      *
-     * @param slot      The slot to adjust the attributes for.
-     * @param modifiers Modifiers currently on the stack either because of a {@link DataComponents#ATTRIBUTE_MODIFIERS} component, or the default attribute modifiers
+     * @param modifiers The default attribute modifiers, either from {@link DataComponent#ATTRIBUTE_MODIFIERS} or {@link #getDefaultAttributeModifiers(ItemStack)}.
      */
-    default void adjustAttributeModifiers(ItemStack stack, EquipmentSlot slot, Multimap<Holder<Attribute>, AttributeModifier> modifiers) {}
+    default ItemAttributeModifiers adjustAttributeModifiers(ItemStack stack, ItemAttributeModifiers modifiers) {
```

Comment:
After further thought, and review of this PR. I agree with you that this hook I added probably should just be removed and we direct people to the event so that we:
a. Only have to convert it to a builder at most once
b. Allow people to make use of the helper methods being added in the event.